frontend/src/metabase/dashboard/reducers.js

```diff
@@ -380,6 +381,15 @@ const loadingDashCards = handleActions(
         };
       },
     },
+    [FETCH_CARD_DATA_PENDING]: {
+      next: (state, { payload: { dashcard_id } }) => {
+        const loadingIds = (state.loadingIds ?? []).concat(dashcard_id);
```

Comment:
we don't need to check if `loadingIds` is undefined because it's guaranteed to not be undefined.
https://github.com/metabase/metabase/blob/35653d30968c1fbe132666ed64194b3992e893ad/frontend/src/metabase/dashboard/constants.ts#L22

I think we should not use guard clause unnecessarily since it implies `loadingIds` could be undefined which could confuse other readers.